support-task/src/test/java/test/camunda/SupportTaskService.java

```diff
@@ -5,5 +5,6 @@
 public interface SupportTaskService {
 
     String createSupportTask(String mainTaskInstanceId, String supportTaskTypeKey, Map<String, Object> variables);
+    void cancelSupportTask(String mainTaskInstanceId, String supportTaskTypeKey, Map<String, Object> variables);
```

Comment:
You should be able to cancel a single instance iidentified by the supportTaskInstanceId, not by type.

Reply:
It is identified by unique combination of maintaskid and taskdef. If a second support task is to be created  by the same main task - it will have another taskdef

Reply:
No it is not. You can have multiple instances of the same support task type.
Know your requirements!!!